pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -258,7 +259,7 @@ func (p *Pod) Stop(ctx context.Context, c client.Client, _ []podset.PodSetInfo,
 	}
 
 	for i := range podsInGroup {
-		if podSuspended(&podsInGroup[i]) || !podsInGroup[i].DeletionTimestamp.IsZero() {
+		if !deletingWorkload && (podSuspended(&podsInGroup[i]) || !podsInGroup[i].DeletionTimestamp.IsZero()) {
```

Comment:
```suggestion
		if !podsInGroup[i].DeletionTimestamp.IsZero() || (!deletingWorkload && podSuspended(&podsInGroup[i]))  {
```

If the Pod already has a deletion timestamp, no need to call delete on it again.

Reply:
Don't we need the new pod condition even if it's been deleted before?

Reply:
Not necessary. The Pod condition should reflect why it was initially deleted. If it wasn't Kueue, there is no need to override.